src/osx/cocoa/dataview.mm

```diff
@@ -2130,6 +2136,37 @@ -(BOOL) becomeFirstResponder
     [m_OutlineView release];
 }
 
+void wxCocoaDataViewControl::keyEvent(WX_NSEvent event, WXWidget slf, void *_cmd)
+{
+    if( [event type] == NSKeyDown && [[event charactersIgnoringModifiers]
+         characterAtIndex: 0] == NSCarriageReturnCharacter )
+    {
+        wxDataViewCtrl* const dvc = GetDataViewCtrl();
+        const wxDataViewItem item = wxDataViewItem( [[m_OutlineView itemAtRow:[m_OutlineView selectedRow]] pointer]);
+        wxDataViewEvent eventDV(wxEVT_DATAVIEW_ITEM_ACTIVATED, dvc, item);
+        if ( !dvc->GetEventHandler()->ProcessEvent(eventDV) )
+            wxWidgetCocoaImpl::keyEvent(event, slf, _cmd);
+    }
+    else
+    {
+        wxWidgetCocoaImpl::keyEvent(event, slf, _cmd);  // all other keys
+    }
+}
+
+bool wxCocoaDataViewControl::doCommandBySelector(void* sel, WXWidget slf, void* _cmd)
+{
+    bool handled = wxWidgetCocoaImpl::doCommandBySelector(sel, slf, _cmd);
+    // if this special key has not been handled
+    if ( !handled && IsInNativeKeyDown() )
```

Comment:
Another question: could we just forward the command to the native `doCommandBySelector:` implementation? I.e. not even check if it's a key down event or not, but just skip everything we don't process here? This would seem to be more future-proof, just in case we can get some commands generated in some other way here.

Again, this is a question/proposal for future enhancement (and I'd also like to see how do you forward the unhandled events generically with Objective C, because I'm not quite sure) and shouldn't prevent this PR from being merged.

Reply:
this doesn't work, I wouldn't need half of the changes if it did :-( apparently NSOutlineView nor its superclasses don't implement this method, they don't implement the NSTextInputClient protocol, therefore I have to feed the original event